api/v1alpha1/ndb_api_helpers.go

```diff
@@ -153,6 +162,115 @@ func GenerateProvisioningRequest(ctx context.Context, ndbclient *ndbclient.NDBCl
+		} else {
+			isValidProfile, matchedProfile, errorThroughChecks := performProfileAvailabilityCheck(ctx, dbEngineSpecificProfiles, softwareProfile, PROFILE_TYPE_SOFTWARE)
+			if errorThroughChecks != nil {
+				//log.Error(err, "")
+				return errorThroughChecks
+			}
+			if isValidProfile {
+				profilesMap[PROFILE_TYPE_SOFTWARE] = matchedProfile
+			}
+		}
+
+		//Custom Compute Profile Check and overriding the default values
+		computeProfile := dbSpec.Instance.Profiles.Compute
+		if computeProfile == (Profile{}) {
+			log.Info("No enrichment for compute profiles as no custom profile received for Compute. Hence, proceeding with default OOB compute profile")
+		} else {
+			isValidProfile, matchedProfile, errorThroughChecks := performProfileAvailabilityCheck(ctx, genericProfiles, computeProfile, PROFILE_TYPE_COMPUTE)
+			if errorThroughChecks != nil {
+				//log.Error(err, "")
+				return errorThroughChecks
+			}
+			if isValidProfile {
+				profilesMap[PROFILE_TYPE_COMPUTE] = matchedProfile
+			}
+		}
+
+		//Custom Network Profile Check and overriding the default values
+		networkProfile := dbSpec.Instance.Profiles.Network
+		if networkProfile == (Profile{}) {
+			log.Info("No enrichment for network profiles as no custom profile received for Network. Hence, proceeding with default OOB network profile")
+		} else {
+			isValidProfile, matchedProfile, errorThroughChecks := performProfileAvailabilityCheck(ctx, dbEngineSpecificProfiles, networkProfile, PROFILE_TYPE_NETWORK)
+			if errorThroughChecks != nil {
+				//log.Error(err, "")
+				return errorThroughChecks
+			}
+			if isValidProfile {
+				profilesMap[PROFILE_TYPE_NETWORK] = matchedProfile
+			}
+		}
+
+		//Custom DbParam Profile Check and overriding the default values
+		dbParamProfile := dbSpec.Instance.Profiles.DbParam
+		if dbParamProfile == (Profile{}) {
+			log.Info("No enrichment for database parameter profiles as no custom profile received for it. Hence, proceeding with default OOB dbParam profile")
+		} else {
+			isValidProfile, matchedProfile, errorThroughChecks := performProfileAvailabilityCheck(ctx, dbEngineSpecificProfiles, dbParamProfile, PROFILE_TYPE_DATABASE_PARAMETER)
+			if err != nil {
+				//log.Error(err, "")
+				return errorThroughChecks
+			}
+			if isValidProfile {
+				profilesMap[PROFILE_TYPE_DATABASE_PARAMETER] = matchedProfile
+			}
+		}
+	}
+	return
+}
+
+func performProfileAvailabilityCheck(ctx context.Context, profiles []ProfileResponse, customProfile Profile, profileType string) (isValidProfile bool, matchedProfile ProfileResponse, err error) {
+	log := ctrllog.FromContext(ctx)
+	isValidProfile = false
+	err = nil
+	matchedProfile = ProfileResponse{}
+	for _, eachProfile := range profiles {
+		if eachProfile.Type == profileType {
+			if (eachProfile.Id == customProfile.Id) && (eachProfile.LatestVersionId == customProfile.VersionId) {
```

Comment:
Also, as discussed earlier.. can we add a method to include profile filtering based on profile names as well?

Reply:
This can be handled in later iterations. Currently, I'm majorly focusing on ID & VersionID
Once, this is tested, extending to name should be quite easy. That's my thinking